### PR TITLE
New locales based on pattern

### LINKED ISSUE #95: UrlGenerator::generate and UrlGenerate::generateAbsolute with nested $parameters arguments.
Good day. After last releases methods `generate` and `generateAbsolute` was failed on that string ` $parameters = array_map('\strval', $parameters);` if send multilevel array like this (generated from Yiisoft\Form\FormModel attributes)
```
[
    'Search' => [
        'address.with_data' => false
        'address.objects.type_ids' => [
            0 => '4'
        ]
        'address.objects.is_active' => true
        'address.pageData.page.source_ids' => [
            0 => '6'
        ]
    ]
]
```

### PR DESCRIPTION
**Pull request by yiiliveext**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

Fix #95

Merge after https://github.com/yiisoft/router/pull/156

### CODE REVIEW

File: src/UrlGenerator.php [hunk 1]
```diff
@@ -50,19 +48,8 @@ public function __construct(
      */
     public function generate(string $name, array $parameters = []): string
     {
-        $parameters = array_map('\strval', $parameters);
+        $parameters = array_map('\strval', array_merge($this->defaults, $parameters));
```
Comment (vjik) [inline src/UrlGenerator.php:51]: For performance better type cast defaults on set.

File: src/UrlGenerator.php [hunk 2]
```diff
@@ -124,6 +111,32 @@ public function generateAbsolute(
...
 
+    public function generateFromCurrent(array $replacedParams, string $fallbackRouteName = null): string
+    {
+        if ($this->currentRoute === null || $this->currentRoute->getName() === null) {
+            if ($fallbackRouteName !== null) {
+                $fallbackRoute = $this->routeCollection->getRoute($fallbackRouteName);
+                return $this->generate($fallbackRouteName, $replacedParams);
+            }
+
+            if ($this->currentRoute->getUri() !== null) {
```
Comment (vjik) [inline src/UrlGenerator.php:122]: Here can be error if `$this->currentRoute` is `null`.

Comment (yiiliveext) [reply src/UrlGenerator.php]: Done

File: src/UrlGenerator.php (hunk 1 above)
Comment (samdark) [reply src/UrlGenerator.php:51]: Should not be much of a performance gain.

File: src/UrlGenerator.php [hunk 3]
```diff
@@ -124,6 +111,31 @@ public function generateAbsolute(
         return $uri === null ? $url : $this->generateAbsoluteFromLastMatchedRequest($url, $uri, $scheme);
     }
 
+    public function generateFromCurrent(array $replacedParams, string $fallbackRouteName = null): string
```
Comment (samdark) [inline src/UrlGenerator.php:114]:
```suggestion
    public function generateFromCurrent(array $replacedParams, ?string $fallbackRouteName = null): string
```

**Pull request merged**